langkit/dsl_unparse.py

```diff
@@ -1,10 +1,177 @@
 from __future__ import absolute_import, division, print_function
 
 from langkit.passes import GlobalPass
+from contextlib import contextmanager
+
+try:
+    import libpythonlang as lpl
+except ImportError:
+    lpl = None
 
 templates = {}
 
 
+class DSLWalker(object):
+    if lpl is not None:
+        ctx = lpl.AnalysisContext()
+
+    class NoOpWalker(object):
+        def __init__(self):
+            pass
+
+        def __getattr__(self, item):
+            return self
+
+        def __call__(self, *args, **kwargs):
+            return self
+
+        def __enter__(self):
+            return self
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            return self
+
+        def emit_comments(self):
+            return ""
+
+    @staticmethod
+    def class_from_location(loc):
```

Comment:
Plz add a docstring

Reply:
There is no docstring, so I assume this was marked as resolved by mistake. ;-)

Reply:
Mh no, there is a docstring! (https://github.com/AdaCore/langkit/pull/275/files#diff-b1a19ccbccaa0cda7fafe777e45fc165R83 -> ctrl+f `class_from_location`)

Reply:
Ah, I see: the comments were added in a later commit: I had the commit view only when I noticed this. Sorry for the noise!